openedx/features/badging/helpers.py

```diff
@@ -0,0 +1,64 @@
+from collections import OrderedDict
+
+from .models import Badge
+
+
+def populate_trophycase(courses, earned_badges):
+    """
+    This method populate trophycase data into dictionary and returns it. This method has a hierarchy of
+    enrolled courses, badge type, earned and yet to be earned badges with complete detail
+    :param courses: Courses enrolled by user
+    :param earned_badges: Badges earned by user
+    :return: dictionary containing trophycase json
+    """
+    trophycase_dict = OrderedDict()
+    badge_queryset = Badge.objects.all().order_by('threshold')
+
+    for course_id, display_name in courses:
+
+        course_badges = get_all_badges(badge_queryset, course_id, earned_badges)
+
+        trophycase_dict[unicode(course_id)] = {
+            'display_name': display_name,
+            'badges': course_badges
+        }
+
+    return trophycase_dict
+
+
+def get_all_badges(badge_queryset, course_id, earned_badges):
```

Comment:
lets make badge_queryset an optional parameter and run the query in case its not passed